pkg/resourceprocessor/diffs.go

```diff
@@ -0,0 +1,107 @@
+package resourceprocessor
+
+import (
+	"fmt"
+	"github.com/kartverket/skiperator/pkg/reconciliation"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (r *ResourceProcessor) getDiff(task reconciliation.Reconciliation) ([]client.Object, []client.Object, []client.Object, []client.Object, error) {
```

Comment:
Future enhancement: something like this?
```suggestion
func (r *ResourceProcessor) getDiff(task reconciliation.Reconciliation) (*resourcesDiff, error) {
```

Reply:
Maybe a struct could work better here? 
Returning 4 lists is very confusing for sure..

```
type Diff struct {
shouldUpdate []client.Object
shouldDelete .......
```

Reply:
added a resourceDiff struct and updated unit tests.